tests/gtests/api/test_memory_desc.cpp

```diff
@@ -88,4 +88,54 @@ HANDLE_EXCEPTIONS_FOR_TEST(memory_desc_test_t, TestComparison) {
     EXPECT_NE(plain_md, blocked_md);
 }
 
+TEST(c_api_host_scalar_md, TestHostScalarF32) {
+    dnnl_memory_desc_t scalar_md = nullptr;
+    DNNL_CHECK(dnnl_memory_desc_create_host_scalar(&scalar_md, dnnl_f32));
+
+    int ndims = -1;
+    dnnl_data_type_t dtype = dnnl_data_type_undef;
+    dnnl_format_kind_t fmt_kind = dnnl_format_kind_undef;
+
+    DNNL_CHECK(dnnl_memory_desc_query(scalar_md, dnnl_query_ndims_s32, &ndims));
+    EXPECT_EQ(ndims, 1);
+
+    DNNL_CHECK(dnnl_memory_desc_query(scalar_md, dnnl_query_data_type, &dtype));
+    EXPECT_EQ(dtype, dnnl_f32);
+
+    DNNL_CHECK(dnnl_memory_desc_query(
+            scalar_md, dnnl_query_format_kind, &fmt_kind));
+    EXPECT_EQ(fmt_kind, dnnl_format_kind_host_scalar);
+
+    size_t sz = dnnl_memory_desc_get_size(scalar_md);
+    EXPECT_EQ(sz, sizeof(float));
+
+    DNNL_CHECK(dnnl_memory_desc_destroy(scalar_md));
+}
```

Comment:
We don't test C API in gtests, C++ only. Same applies to Memory test.

Reply:
> Is there any particular reason you need me to rewrite those C API tests to C++?

Nobody uses C API.
`EXPECT_ANY_THROW`, `EXPECT_NO_THROW` will help with convenience.

Reply:
I will ensure that C++ API coverage for the new APIs is the same as for C APIs, but I will not be removing those, as we do have testing for C APIs in gtests, and we do have C APIs in the library, so there is still a need for interface tests.